packages/tailwindcss/src/variants.ts

```diff
@@ -95,12 +101,11 @@ export class Variants {
       name,
       (r) => {
         let body = structuredClone(ast)
+        if (usesAtVariant) substituteAtVariant(body, designSystem)
```

Comment:
This is technically evaluated lazily and every time this function is called instead of ahead of time.

But once we compute a variant we cache it anyway so should be fine.